File: template/app/src/main/java/com/example/base/baseapp/dagger/ActivityComponent.java
package com.example.base.baseapp.dagger;

import com.example.base.baseapp.dagger.scope.ActivityScope;
import com.example.base.baseapp.module.main.MainActivity;

import dagger.Subcomponent;

@Subcomponent(modules = {})
@ActivityScope
public interface ActivityComponent {
    void inject(MainActivity activity);
}

File: generators/app/templates/app/src/main/java/com/example/base/baseapp/dagger/ActivityComponent.kt
package com.example.base.baseapp.dagger

import com.capgemini.templateapp.dagger.scope.ActivityScope
import com.example.base.baseapp.modules.main.MainActivity

import dagger.Subcomponent

@Subcomponent(modules = [])
@ActivityScope
interface ActivityComponent : BaseActivityComponent {
    fun inject(activity: MainActivity)
}
